### PR TITLE
First attempt adding browser-bunyan types

### PR DESCRIPTION
**Pull request by PaulLockwood**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `npm run new-package package-name`, not by basing it on an existing project.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: browser-bunyan/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (andy-ms) [inline browser-bunyan/package.json]: You don't need this file. `package.json` is just for [special cases](https://github.com/DefinitelyTyped/DefinitelyTyped#i-notice-some-packages-having-a-packagejson-here).

File: browser-bunyan/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es6",
+        "noImplicitAny": true,
+        "noImplicitThis": false,
```
Comment (andy-ms) [inline browser-bunyan/tsconfig.json]: Make these `true`.

File: browser-bunyan/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
+// Type definitions for browser-bunyan 
+// Project: https://github.com/philmander/browser-bunyan
+// Definitions by: Paul Locwood <https://github.com/PaulLockwood> 
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// essentially a copy for type definitions for node-bunyan with the module rename
```
Comment (andy-ms) [inline browser-bunyan/index.d.ts]: Could this whole file just be `export * from "bunyan";`?

File: browser-bunyan/browser-bunyan-tests.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+var bunyan = require('browser-bunyan');
```
Comment (andy-ms) [inline browser-bunyan/browser-bunyan-tests.ts]: Needs TypeScript import syntax (`import * as bunyan from "browser-bunyan"`) or this file won't be typechecked.

**Review by ghost — COMMENTED** [review]
Needs a `tslint.json` containing `{ "extends": "../tslint.json" }`

File: browser-bunyan/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es6",
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (andy-ms) [inline browser-bunyan/tsconfig.json]: Make this `true`.

Comment (PaulLockwood) [inline browser-bunyan/tsconfig.json]: I could not get your export suggestion to work with this DefinitelyTyped Repo.

Below works in our local project. It nor your suggestion (from "bunyan") would not pass Travis CI.

```
  export * from "@types/bunyan
```

**Comment by PaulLockwood** [comment]
It should be ready to be merged in now.

**Comment by PaulLockwood** [comment]
Do I need to do anything else? This is my first Pull Request.

File: browser-bunyan/tsconfig.json (hunk 5 above)
Comment (andy-ms) [inline browser-bunyan/tsconfig.json]: Looking at the [error message](https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/188708952#L1489), that was a lint error. Completely unrelated to `export *`. Could you try it again and run the linter locally to make sure you have no lint errors?

**Comment by PaulLockwood** [comment]
There are no linting errors in the .d.ts file. There are semicolons missing in the tests (from the library these types are for and I'd rather not alter the tests at all).

Re-running locally no linting errors are detected:

```
npm run lint -- browser-bunyan

> definitely-typed@0.0.1 lint /Users/paullockwood/Code/DefinitelyTyped
> node scripts/lint.js "browser-bunyan"

node ../node_modules/tslint/lib/tslint-cli --format stylish "**/*.d.ts"

```

File: browser-bunyan/tsconfig.json (hunk 5 above)
Comment (PaulLockwood) [inline browser-bunyan/tsconfig.json]: There are no linting errors in the .d.ts file. There are semicolons missing in the tests (from the library these types are for and I'd rather not alter the tests at all).

Re-running locally no linting errors are detected:

```
npm run lint -- browser-bunyan

> definitely-typed@0.0.1 lint /Users/paullockwood/Code/DefinitelyTyped
> node scripts/lint.js "browser-bunyan"

node ../node_modules/tslint/lib/tslint-cli --format stylish "**/*.d.ts"

```

Comment (PaulLockwood) [inline browser-bunyan/tsconfig.json]: The build you linked to is not the latest. [This is the latest Travis CI run](https://travis-ci.org/PaulLockwood/DefinitelyTyped/builds/188718862) (which passed).

Comment (andy-ms) [inline browser-bunyan/tsconfig.json]: Yes, I was referring to the older version with `export * from "bunyan";`.

Comment (PaulLockwood) [inline browser-bunyan/tsconfig.json]: Hooray! Is it good to merge in?

Comment (andy-ms) [inline browser-bunyan/tsconfig.json]: You should try `export * from "bunyan";` again. You mentioned that it failed the first time, but that failure was actually an unrelated lint error.

Comment (PaulLockwood) [inline browser-bunyan/tsconfig.json]: They were different errors, but will try it again. Do you think it should be:

```
export * from "bunyan";
```
or
```
export * from "@types/bunyan";
```

Comment (andy-ms) [inline browser-bunyan/tsconfig.json]: `export * from "bunyan";`.

Comment (PaulLockwood) [inline browser-bunyan/tsconfig.json]: Neat, you were correct and Travis CI says green.

**Comment by PaulLockwood** [comment]
Cheers, and thanks for the assistance.

**Pull request merged**